### PR TITLE
refactor(nexus): remove unnecessary code from artifact downloads

### PR DESCRIPTION
**Pull request by ibindlish**
Description
-----------

- Fixes [WB-16257](https://wandb.atlassian.net/browse/WB-16257)

What does the PR do?

Based on post-merge review comments on https://github.com/wandb/wandb/pull/6275
- Remove code around setting the `DefaultDownloadRoot` for an artifact that we're downloading as the `downloadRoot` is already set by the python user process. 
- Fix potential nil-pointer dereferences while reading gql response objects.
- Change the input for download_artifact to use `artifact_id` instead of `qualified_name`
 

### <samp>🤖 Generated by Copilot at 18352e8</samp>

No summary available (Limit exceeded: required to process 51797 tokens, but only 50000 are allowed per call)

Testing
-------

```
wandb.require("nexus")
with wandb.init() as run:
    artifact = run.use_artifact('test-paige/uncategorized/b2:v0', type='tmp')
    path = artifact.download()
    print(f"\n\n Path: {path}")
```

copilot:poem

[WB-16257]: https://wandb.atlassian.net/browse/WB-16257?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: nexus/pkg/artifacts/manifest.go [hunk 1]
```diff
@@ -86,9 +88,28 @@ func (m *Manifest) GetManifestEntryFromArtifactFilePath(path string) (ManifestEn
...
 	manifestEntry, ok := manifestEntries[path]
 	if !ok {
-		// implement _get_obj_entry
-		// For now just return error
 		return ManifestEntry{}, fmt.Errorf("path not contained in artifact: %s", path)
 	}
 	return manifestEntry, nil
 }
+
+func loadManifestFromURL(url string) (Manifest, error) {
```
Comment (ibindlish) [inline nexus/pkg/artifacts/manifest.go:96]: moved here from `downloader.go`

File: wandb/sdk/artifacts/artifact.py [hunk 2]
```diff
@@ -1755,8 +1755,8 @@ def _download(
     ) -> FilePathStr:
         # todo: remove once artifact reference downloads are supported in nexus
         require_nexus = False
```
Comment (szymon-piechowicz-wandb) [inline wandb/sdk/artifacts/artifact.py:1757]: Nit: delete, this is never used.

**Review by szymon-piechowicz-wandb — APPROVED** [review]

File: nexus/pkg/artifacts/downloader.go [hunk 3]
```diff
@@ -25,96 +21,70 @@ type ArtifactDownloader struct {
...
+		return gql.ArtifactByIDArtifact{}, err
 	}
-	return *response.GetProject().GetArtifact(), nil
-}
-
-func loadManifestFromURL(url string) (Manifest, error) {
-	resp, err := http.Get(url)
-	if err != nil {
-		return Manifest{}, err
+	artifact := response.GetArtifact()
```
Comment (szymon-piechowicz-wandb) [inline nexus/pkg/artifacts/downloader.go:62]: Nit:
```suggestion
	artifact := response.Artifact
```

File: nexus/pkg/artifacts/downloader.go [hunk 4]
```diff
@@ -25,96 +21,70 @@ type ArtifactDownloader struct {
...
+	return *artifact, nil
 }
 
-func (ad *ArtifactDownloader) getArtifactManifest() (artifactManifest Manifest, rerr error) {
-	entityName, projectName, artifactName, err := parseArtifactQualifiedName(ad.QualifiedName)
-	if err != nil {
-		return Manifest{}, err
-	}
+func (ad *ArtifactDownloader) getArtifactManifest(artifact gql.ArtifactByIDArtifact) (manifest Manifest, rerr error) {
 	response, err := gql.ArtifactManifest(
```
Comment (szymon-piechowicz-wandb) [inline nexus/pkg/artifacts/downloader.go:70]: Nit: we can fetch the artifact by id in this query, and then we won't need `ArtifactByID` at all.

**Review by dmitryduev — APPROVED** [review]
👍🏻

File: nexus/pkg/artifacts/downloader.go [hunk 5]
```diff
@@ -284,28 +227,18 @@ func (ad *ArtifactDownloader) downloadFiles(artifactID string, manifest Manifest
...
-	if ad.DownloadRoot == nil {
-		err = ad.setDefaultDownloadRoot(artifactAttrs)
-		if err != nil {
-			return "", err
-		}
-		downloadRoot = *ad.DefaultDownloadRoot
-	} else {
-		downloadRoot = *ad.DownloadRoot
+		return err
 	}
```
Comment (szymon-piechowicz-wandb) [inline nexus/pkg/artifacts/downloader.go:234]: Remove.

**Pull request merged**